tests/console/java.pm

```diff
@@ -37,20 +38,33 @@ sub run {
     # https://www.suse.com/releasenotes/x86_64/SUSE-SLES/15-SP2/#development-java-versions
     # java-11-openjdk                   -> Basesystem
     # java-10-openjdk & java-1_8_0-ibm  -> Legacy
-    my $cmd = 'install --auto-agree-with-licenses ';
-    $cmd .= (is_sle('15+') || is_leap) ? 'java-11-openjdk* java-1_*' : 'java-*';
-    zypper_call($cmd, timeout => 2000);
-
-    if (script_run 'rpm -q wget') {
-        zypper_call 'in wget';
+    if (check_var('TRANSACTIONAL_SERVER', '1')) {
+        select_console 'root-console';
+        assert_script_run("transactional-update --continue pkg install --no-confirm java-*", 2000);
+        check_reboot_changes;
+        reset_consoles;
+        select_console('root-console', 200);
+        assert_script_run 'wget --quiet ' . data_url('console/test_java_transactional-server.sh');
+        assert_script_run 'chmod +x test_java_transactional-server.sh';
+        assert_script_run './test_java_transactional-server.sh';
+        record_info("issue", "read-only file system, see poo#77716");
     }
-    assert_script_run 'wget --quiet ' . data_url('console/test_java.sh');
-    assert_script_run 'chmod +x test_java.sh';
-    assert_script_run './test_java.sh';
-    # if !QAM test suite then cleanup test suite environment
-    unless (is_updates_tests || is_opensuse) {
-        remove_suseconnect_product('sle-module-legacy');
-        (script_run 'rpm -qa | grep java-1_') || zypper_call('rm java-1_*');
+    else {
+        my $cmd = 'install --auto-agree-with-licenses ';
+        $cmd .= (is_sle('15+') || is_leap) ? 'java-11-openjdk* java-1_*' : 'java-*';
+        zypper_call($cmd, timeout => 2000);
+
+        if (script_run 'rpm -q wget') {
+            zypper_call 'in wget';
```

Comment:
As you'll probably have to introduce something like `my $packagelist = (is_sle('15+') || is_leap) ? 'java-11-openjdk* java-1_*' : 'java-*';` anyway, you could also add `wget` to that unconditionally and drop this block here.

Reply:
we need to keep this for wget because it needs to be installed on jeos:  http://10.162.23.47/tests/8528#step/java/12
I don't break tests :)

Reply:
What I meant was to add `wget` to the list of `java` packages instead of doing it conditionally here.